### PR TITLE
fix: cleanup. Readme

### PR DESCRIPTION
**Pull request by KodiVerse**

### CODE REVIEW

File: src/module/dy/create-model.js [hunk 1]
```diff
@@ -77,10 +77,10 @@ module.exports = (kwargs) => {
       }))
     }),
     BillingMode: 'PAY_PER_REQUEST'
-  };
+  });
 
   return {
-    schema,
+    schema: genSchema(),
```
Comment (KodiVerse) [inline src/module/dy/create-model.js]: This is not the best. Are we just going to pass the handler all the way through? If we do I need to write some more tests for coverage

File: README.md [hunk 2]
```diff
@@ -133,6 +133,42 @@ limit is reached or no more keys are available. Uses [s3:listObjectsV2](https://
 Returns a non-ASCII key representation for an encoded s3 key. Useful to obtain the
 not-encoded key representation after calling `listObjects`.
 
+#### dy.Model({ name: String, attributes: Object, indices: Object })
+Instantiates Model.<br>
+Internally uses [dynamodb-toolbox](https://github.com/jeremydaly/dynamodb-toolbox)
+
+#### dy.Model.upsert(item: Object, opts: Object)
```
Comment (simlu) [inline README.md]:
```suggestion
##### dy.Model().upsert(item: Object, opts: Object)
```

File: README.md [hunk 3]
```diff
@@ -133,6 +133,42 @@ limit is reached or no more keys are available. Uses [s3:listObjectsV2](https://
...
+Internally uses [update](https://github.com/jeremydaly/dynamodb-toolbox#updatekey-options-parameters)
+
+#### dy.Model.getItemOrNull(key: String, opts: Object)
+Returns entry or null if not found.<br>
+Options include:
+- `toReturn` (Array): Fields to return.
+Internally uses [get](https://github.com/jeremydaly/dynamodb-toolbox#getkey-options-parameters)
+
+#### dy.Model.query(key: String, opts: Object)
+Pages through table based on primary key values.<br>
```
Comment (simlu) [inline README.md:160]:
```suggestion
Pages through table based on primary key values.<br>
```
Is that true? I think it's whatever index you pass and if no index is passed it uses the primary index

File: src/module/dy/create-model.js (hunk 1 above)
Comment (simlu) [reply src/module/dy/create-model.js]: I'm not sure I understand?

Comment (simlu) [reply src/module/dy/create-model.js]: pass genSchema() through everywhere?

File: src/module/dy.js [hunk 4]
```diff
@@ -28,12 +27,14 @@ module.exports = ({ call, getService, logger }) => ({
...
         returnValues = 'all_new',
-        conditions = null
+        conditions: updateConditions = null
       } = {}) => {
-        const result = await model.entity.update(item, {
-          returnValues,
-          ...(conditions === null ? {} : { conditions })
-        });
+        const conditions = model.schema.KeySchema.reduce((prev, { AttributeName: attr }) => prev
+          .concat({ attr, exists: true }), []);
```
Comment (simlu) [inline src/module/dy.js]: do a map instead

File: src/module/dy/create-model.js [hunk 5]
```diff
@@ -77,10 +77,19 @@ module.exports = (kwargs) => {
       }))
     }),
     BillingMode: 'PAY_PER_REQUEST'
-  };
+  });
 
   return {
-    schema,
+    genSchema,
+    getSchema: (() => {
```
Comment (simlu) [inline src/module/dy/create-model.js]:
```suggestion
    get schema (() => {
```

File: src/module/dy/create-model.js [hunk 6]
```diff
@@ -77,10 +77,18 @@ module.exports = (kwargs) => {
...
-    schema,
+    get schema() {
+      let cache;
+      return () => {
+        if (cache === undefined) {
+          cache = genSchema();
+        }
+        return cache;
+      };
+    },
```
Comment (simlu) [inline src/module/dy/create-model.js]:
```suggestion
    get schema() => {
        if (cache === undefined) {
          cache = genSchema();
        }
        return cache;
      },
```

File: src/module/dy/create-model.js [hunk 7]
```diff
@@ -22,6 +22,8 @@ const convertType = (t) => {
   }
 };
 
+let cache;
```
Comment (simlu) [inline src/module/dy/create-model.js]: nononononono

**Review by simlu — APPROVED** [review]

**Pull request merged**